tensorflow/contrib/tensorrt/test/tf_trt_integration_test_base.py

```diff
@@ -514,18 +525,24 @@ def _Test(self):
 
   use_optimizer_options = [False, True]
   dynamic_engine_options = [False, True]
-  for (use_optimizer, precision_mode, dynamic_engine) in itertools.product(
-      use_optimizer_options, PRECISION_MODES, dynamic_engine_options):
+  use_calibration_options = [False, True]
+  opts = itertools.product(use_optimizer_options, PRECISION_MODES,
+                           dynamic_engine_options, use_calibration_options)
+  for (use_optimizer, precision_mode, dynamic_engine, use_calibration) in opts:
     if IsQuantizationMode(precision_mode):
       if use_optimizer:
         # TODO(aaroey): if use_optimizer is True we need to get the inference
         # graphdef using custom python wrapper class, which is not currently
         # supported yet.
         continue
-      if not dynamic_engine:
+      if not dynamic_engine and use_calibration:
```

Comment:
Why changing this line? IIRC, the original condition is "run INT8 conversion if it's dynamic engine", the new one is "run INT8 conversion if it's dynamic engine or if it's static engine without calibration". I think we don't want to run conversion with static engine which is not supported yet.

Reply:
Oh right, thanks for catching that! My intention here was to run the following:
int8=1, dynamic, use_calibration=1
int8=1, dynamic, use_calibration=0
~~int8=1, static, use_calibration=1~~ <- exclude this
int8=1, static, use_calibration=0

So the correct logic should be
`if not dynamic_engine or not use_calibration`
Does that seem right?

Reply:
INT8+static engine will fall back to dynamic engine, see [here](https://github.com/tensorflow/tensorflow/pull/22788/files#diff-63b6c12e8d9c0da3f1a4476c60494ea5R910), is that what you want to test?

Reply:
>```curr_engine.engine_type = ((params.is_dyn_op || params.use_calibration)
             ? EngineInfo::EngineType::TRTDynamic
             : EngineInfo::EngineType::TRTStatic);```

use_calibration+static engine should fall back to dynamic engine, but not use_calibration+static engine will be static

Reply:
I see, make sense. Thanks.

Reply:
Thanks, sorry for the confusion

Reply:
I read this one more time and is confused again until read through the discussions here. So please add a comment saying the fact that, !use_calibration+static engine will be static, and we want to test that.